src/mmw/js/src/core/models.js

```diff
@@ -136,7 +150,7 @@ var TaskModel = Backbone.Model.extend({
             }),
             pollingDefer = $.Deferred();
 
-            startDefer.done(function() {
+            startDefer.done(handleNullInCallback(function() {
```

Comment:
>Where do these null references come from in the first place?

They occur because UI objects that existed when the async process starts have been destroyed by the time the callbacks are invoked.

>Replace done with then. If done fails I think nothing happens, but if then fails it gets caught and goes to fail

I think you're probably right that this could be done within a `then()` but it may not look that different. I think the two options in that case are either `then()` would chain the exception handler into the promise and the `done` and `fail` callbacks are chained unmodified; or, specify the `done` and `fail` (wrapped) callbacks in `then` and then have a single resolution. I'm not sure if an exception raised within a callback within `then()` would result in the chained `fail()` being invoked, but assuming it would, it wouldn't be the desired behavior because `fail` would then be invoked *either* because the promise returned successfully but the callback failed, or because the promise failed to begin with.